sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -290,8 +290,10 @@ int _fopts_mlme_link_check_req(lorawan_buffer_t *buf)
     return GNRC_LORAWAN_CID_SIZE;
 }
 
-static void _mlme_link_check_ans(gnrc_lorawan_t *mac, uint8_t *p)
+static int _mlme_link_check_ans(gnrc_lorawan_t *mac, uint8_t *p, size_t len, uint8_t index)
 {
+    (void) len;
+    (void) index;
     mlme_confirm_t mlme_confirm;
 
     mlme_confirm.link_req.margin = p[1];
```

Comment:
you didn't update this. This will always read the second byte of the fopts buffer, regardless of `index`. And since we are there, maybe this should check that the size is enough and it doesn't overflow. We can return 0 to indicate there was an error and catch it in the caller.